packages/authentication/test/poc/mock-express-router.js

```diff
@@ -0,0 +1,79 @@
+var fs = require('fs');
+var parser = require('url');
+var pathRegexp = require('path-to-regexp');
+
+var router = exports = module.exports = {};
+
+router.stack = [];
+
+router.route = function(path, method) {
```

Comment:
Thanks. I think the part that confuses me most is why are you mocking Express features like `res.send` and the router when none of that is required by Passport strategies AFAICT.

I checked the source code of several Passport Strategies and none of them is using `res.send`. I believe the only place using that method is your own code:

https://github.com/strongloop/loopback-next/pull/192/files#diff-4448f8e203908362282bb6d4a91e2a48R17
```js
app.get('/private', 
  passport.authenticate('basic', { session: false }),
  function(req, res) {
    res.json(req.user); // <-- HERE
  });
```

I cannot comment on the usage of `request.session`, I don't know enough about that.

FWIW, here is how koa-passport mocks required Express features:
 - Request - https://github.com/rkusa/koa-passport/blob/7ae875056cc77206bc44e23f197e394163245185/lib/framework/request.js
 - Response - https://github.com/rkusa/koa-passport/blob/7ae875056cc77206bc44e23f197e394163245185/lib/framework/koa.js#L111-L128

> Without a working model, how do you suggest to bridge passport strategies to loopback-next ?

That's a good question, indeed. I agree with your approach where you to mocked certain express features to support Passport Strategy requirements. My concern is that you are mocking/stubbing things that are not needed by Passport strategies and I would rather not have those unnecessary mocks landed in loopback-next.